tf/net.py

```diff
@@ -0,0 +1,191 @@
+
+
+    def get_weights(self):
+        """Returns the weights as floats per layer"""
+        if self.weights == []:
+            self.denorm_layer(self.pb.weights.ip2_val_b, self.weights)
+            self.denorm_layer(self.pb.weights.ip2_val_w, self.weights)
+            self.denorm_layer(self.pb.weights.ip1_val_b, self.weights)
+            self.denorm_layer(self.pb.weights.ip1_val_w, self.weights)
+            self.denorm_conv_block(self.pb.weights.value, self.weights)
+
+            self.denorm_layer(self.pb.weights.ip_pol_b, self.weights)
+            self.denorm_layer(self.pb.weights.ip_pol_w, self.weights)
+            self.denorm_conv_block(self.pb.weights.policy, self.weights)
+
+            for res in reversed(self.pb.weights.residual):
+                self.denorm_conv_block(res.conv2, self.weights)
+                self.denorm_conv_block(res.conv1, self.weights)
+
+            self.denorm_conv_block(self.pb.weights.input, self.weights)
+            
+        return self.weights
+
+
+    def filters(self):
+        w = self.get_weights()
+        return len(w[1])
+
+
+    def blocks(self):
+        w = self.get_weights()
+        blocks = len(w) - (4 + 14)
+
+        if blocks % 8 != 0:
+            raise ValueError("Inconsistent number of weights in the file")
+
+        return blocks // 8
+
+
+    def parse_proto(self, filename):
+        with gzip.open(filename, 'rb') as f:
+            self.pb = self.pb.FromString(f.read())
+
+
+    def parse_txt(self, filename):
+        weights = []
+
+        with open(filename, 'r') as f:
+            f.readline()
+            for e, line in enumerate(f):
+                weights.append(list(map(float, line.split(' '))))
+
+        self.fill_net(weights)
+
+
+    def fill_net(self, weights):
+        self.weights = []
+        filters = len(weights[1])
+        blocks = len(weights) - (4 + 14)
+
+        if blocks % 8 != 0:
+            raise ValueError("Inconsistent number of weights in the file")
+        blocks //= 8
+
+        self.pb.format.weights_encoding = pb.Format.LINEAR16
+        self.fill_layer(self.pb.weights.ip2_val_b, weights)
+        self.fill_layer(self.pb.weights.ip2_val_w, weights)
+        self.fill_layer(self.pb.weights.ip1_val_b, weights)
+        self.fill_layer(self.pb.weights.ip1_val_w, weights)
+        self.fill_conv_block(self.pb.weights.value, weights)
+
+        self.fill_layer(self.pb.weights.ip_pol_b, weights)
+        self.fill_layer(self.pb.weights.ip_pol_w, weights)
+        self.fill_conv_block(self.pb.weights.policy, weights)
+
+        tower = []
+        for i in range(blocks):
+            tower.append(self.pb.weights.residual.add())
+
+        for res in reversed(tower):
+            self.fill_conv_block(res.conv2, weights)
+            self.fill_conv_block(res.conv1, weights)
+
+        self.fill_conv_block(self.pb.weights.input, weights)
+
+
+def main(argv):
+    net = Net()
+
+    if argv.input.endswith(".txt"):
```

Comment:
Any thoughts on using auto detect here rather than file name extension?

Reply:
we could, but this is solely for testing purposes so I don't think it's required. lc0 will support both types anyway.